src/xapiwrapper.js

```diff
@@ -1481,99 +1659,48 @@ function isDate(date) {
-        if (!xDomainRequest) {
-            urlPort = (urlparts[3] === null ? ( urlparts[1] === 'http' ? '80' : '443') : urlparts[3]);
-            xDomainRequest = (urlPort === location.port);
-        }
-
         //Add extended LMS-specified values to the URL
         if (lrs !== null && lrs.extended !== undefined) {
-            extended = new Array();
-            for (prop in lrs.extended) {
+            var extended = new Array();
+            for (var prop in lrs.extended) {
                 extended.push(prop + "=" + encodeURIComponent(lrs.extended[prop]));
             }
             if (extended.length > 0) {
                 url += (url.indexOf("?") > -1 ? "&" : "?") + extended.join("&");
             }
         }
-        
-        //If it's not cross domain or we're not using IE, use the usual XmlHttpRequest
-        var windowsVersionCheck = window.XDomainRequest && (window.XMLHttpRequest && new XMLHttpRequest().responseType === undefined);
-        if (!xDomainRequest || windowsVersionCheck === undefined || windowsVersionCheck===false) {
-            xhr = new XMLHttpRequest();
-            xhr.withCredentials = withCredentials; //allow cross domain cookie based auth
-            xhr.open(method, url, callback != null);
-            for(var headerName in headers){
-                xhr.setRequestHeader(headerName, headers[headerName]);
-            }
-        }
-        //Otherwise, use IE's XDomainRequest object
-        else {
-            ieXDomain = true;
-            ieModeRequest = ie_request(method, url, headers, data);
-            xhr = new XDomainRequest();
-            xhr.open(ieModeRequest.method, ieModeRequest.url);
-        }
 
-        //Setup request callback
-        function requestComplete() {
-            if(!finished){
-                // may be in sync or async mode, using XMLHttpRequest or IE XDomainRequest, onreadystatechange or
-                // onload or both might fire depending upon browser, just covering all bases with event hooks and
-                // using 'finished' flag to avoid triggering events multiple times
-                finished = true;
-                var notFoundOk = (ignore404 && xhr.status === 404);
-                if (xhr.status === undefined || (xhr.status >= 200 && xhr.status < 400) || notFoundOk) {
-                    if (callback) {
-                        if(callbackargs){
-                            strictCallbacks ? callback(null, xhr, callbackargs) : callback(xhr, callbackargs);
-                        }
-                        else {
-                          var body;
+        var async = Boolean(callback)
 
-                            try {
-                                body = JSON.parse(xhr.responseText);
-                            }
-                            catch(e){
-                                body = xhr.responseText;
-                            }
+        var result;
+        http(
```

Comment:
This is where the new `http` function is used, returning the `xhr` object as produced.
It greatly simplifies the `XHR_request` function to abstract the `xhr` production but it isn't necessary that this change happen for the node `XMLHttpRequest` shim to work.